llvm/test/CodeGen/DirectX/atan2.ll

```diff
@@ -0,0 +1,52 @@
+; RUN: opt -S -dxil-op-lower -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
+
+; Make sure correct dxil expansions for atan2 are generated for float and half.

```

Comment:
i personally also like to have expansion only tests.

Reply:
I added the expansion-only test.  I experimented with -scalarizer for vector tests, but since this just expands to atan and other instructions, it's just testing scalarization of the atan intrinsic, which is already tested elsewhere.  I can add a vector expansion-only test, if you think that's useful (I have one, but didn't add it here yet).

Reply:
so  if you don't add a vector test then you can remove the `-scalarizer` flag.  

While a vector test would overlap with  the `atan` vector tests. Its benefit is that this unit tests could also be considered an integration test.

Its a small benefit so I don't have a strong opinion here since we have been using `opt` to do our tests. That said I'll signal here that I think long term it would better for us to use `llc` for tests cases like this one where we want to see the final transformation and not just the per pass transformations.

Reply:
I added a vector test.  Let me know if this looks good now.